File: src/CourseObject/CourseSession.java
package CourseObject;

import PersonObject.*;

import java.util.ArrayList;

public class CourseSession extends Course {

	private Id sessionId;

	public CourseSession(Id courseID, String courseName, int quota, int term, Lecturer lecturer,
						 Id sessionId, ArrayList<CourseSchedule> courseSchedules, int credit, CourseType courseType) {
		super(courseID, courseName, quota, term, lecturer, courseSchedules,credit, courseType);
		this.sessionId = sessionId;
	}
	@Override
	public boolean equals(Object o){
		if(o instanceof CourseSession){
			return ((CourseSession) o).getCourseId().getId().equals(this.getCourseId().getId()) && ((CourseSession) o).getSessionId().getId().equals(this.getSessionId().getId());
		}
		return false;
	}
	public Id getSessionId() {
		return sessionId;
	}
}

File: python_development/CourseSession.py
from python_development.Course import Course


class CourseSession(Course):
    def __init__(self, courseID, courseName, quota, term, lecturer, sessionId, courseSchedules, credit, courseType):
        super().__init__(courseID, courseName, quota, term, lecturer, courseSchedules, credit, courseType)
        self.sessionId = sessionId

    def equals(self, obj):
        if isinstance(obj, CourseSession):
            return (
                    obj.getCourseId().getId() == self.getCourseId().getId()
                    and obj.getSessionId().getId() == self.getSessionId().getId()
            )
        return False


    def getSessionId(self):
        return self.session_id
